k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -3022,7 +2995,8 @@ public void onMessageBuildCancel() {
     @Override
     public void onMessageBuildException(MessagingException me) {
         Log.e(K9.LOG_TAG, "Error sending message", me);
-        Toast.makeText(MessageCompose.this, getString(R.string.send_aborted), Toast.LENGTH_LONG).show();
+        Toast.makeText(MessageCompose.this,
+                getString(R.string.send_failed_reason, me.getLocalizedMessage()), Toast.LENGTH_LONG).show();
         currentMessageBuilder = null;
```

Comment:
Right. We already have a class for openpgp api errors, which also does error codes. We don't really use that in OpenKeychain yet but I have some ideas how to do that from our internal error logs. I also wanted to pass a pending intent result which shows the log we keep internally, which may be helpful for debugging.

Anyways, I reckon I'll wait with that until there is better error handling in k9 first.